From 819e481f7909f5d716c0aefb18c83230496a1765 Mon Sep 17 00:00:00 2001
From: guoguo <geekguo@example.com>
Date: Sun, 19 Dec 2021 22:12:45 +0800
Subject: [PATCH 1/2] fix ar8216 ethernet_packet_mangle

---
 drivers/net/phy/ar8216.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/net/phy/ar8216.c b/drivers/net/phy/ar8216.c
index a06d5dd1a..4cf269f25 100644
--- a/drivers/net/phy/ar8216.c
+++ b/drivers/net/phy/ar8216.c
@@ -519,6 +519,8 @@ ar8216_read_port_link(struct ar8xxx_priv *priv, int port,
 	}
 }
 
+#ifdef CONFIG_ETHERNET_PACKET_MANGLE
+
 static struct sk_buff *
 ar8216_mangle_tx(struct net_device *dev, struct sk_buff *skb)
 {
@@ -585,6 +587,8 @@ ar8216_mangle_rx(struct net_device *dev, struct sk_buff *skb)
 	buf[15 + 2] = vlan & 0xff;
 }
 
+#endif
+
 int
 ar8216_wait_bit(struct ar8xxx_priv *priv, int reg, u32 mask, u32 val)
 {
@@ -2455,6 +2459,7 @@ ar8xxx_phy_config_init(struct phy_device *phydev)
 	if (ret)
 		return ret;
 
+#ifdef CONFIG_ETHERNET_PACKET_MANGLE
 	/* VID fixup only needed on ar8216 */
 	if (chip_is_ar8216(priv)) {
 		dev->phy_ptr = priv;
@@ -2462,6 +2467,7 @@ ar8xxx_phy_config_init(struct phy_device *phydev)
 		dev->eth_mangle_rx = ar8216_mangle_rx;
 		dev->eth_mangle_tx = ar8216_mangle_tx;
 	}
+#endif
 
 	return 0;
 }
@@ -2690,10 +2696,12 @@ ar8xxx_phy_detach(struct phy_device *phydev)
 	if (!dev)
 		return;
 
+#ifdef CONFIG_ETHERNET_PACKET_MANGLE
 	dev->phy_ptr = NULL;
 	dev->priv_flags &= ~IFF_NO_IP_ALIGN;
 	dev->eth_mangle_rx = NULL;
 	dev->eth_mangle_tx = NULL;
+#endif
 }
 
 static void
-- 
2.25.1

